### PR TITLE
Added valid_move? method to King model

### PR DESCRIPTION
**Pull request by snsmith92**
Hi team! Feedback on this model for the king piece?

### CODE REVIEW

File: app/models/king.rb [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
 
+  def valid_move?(piece_destination)
+    piece[0] = x_current
+    piece[1] = y_current
+    piece_destination[0] = x_destination
+    piece_destination[1] = y_destination
 
-end
+    # If piece_destination is not within the board -- not allowed
+    if x_destination < 1 || x_destination > 8 || y_destination < 1 || y_destination > 8
```
Comment (ilyakrasnov) [inline app/models/king.rb]: we could make this a separate method and call it sth like `if outside_board(x_dest, y_dest)`..

**Comment by ilyakrasnov** [comment]
@snsmith92 Looks good - just added one refactoring suggestion 👍

**Comment by snsmith92** [comment]
Thanks @ilyakrasnov -- I changed it! With your approval I'll merge.

File: app/models/piece.rb [hunk 2]
```diff
@@ -4,46 +4,52 @@ class Piece < ApplicationRecord
 
   #needs models that inherit from piece called pawn, rook, knight, bishop, queen, king
 
+  def outside_board?(position_x, position_y)
+    if position_x < 0 || position_x > 7 || position_y < 0 || position_y > 7
```
Comment (ilyakrasnov) [inline app/models/piece.rb]: This method can be just one line without `if/else`.. 
```
def
 a || b || c || d
end
```

the condition will evaluate to true and false and be returned as in ruby the last line of the method always gets returned :)

**Comment by ilyakrasnov** [comment]
Left one more suggestion @snsmith92 you're good to merge after it 👍

**Comment by snsmith92** [comment]
Great!

**Pull request merged**